src/NJsonSchema.CodeGeneration.CSharp/CSharpTypeResolver.cs

```diff
@@ -70,7 +70,8 @@ public string Resolve(JsonSchema schema, bool isNullable, string typeNameHint, b
             // Primitive schemas (no new type)
             if (Settings.GenerateOptionalPropertiesAsNullable &&
                 schema is JsonSchemaProperty property &&
-                !property.IsRequired)
+                !property.IsRequired &&
+                !schema.IsEnumeration)
```

Comment:
This doesnt look right, because an enum can also be nullable - or am I missing something?